### PR TITLE
Add ci() methods for condition handling

### LINKED ISSUE #76: Deploy step: Condition handling
```yml
deploy:
  provider: script
  script: R -q -e 'tic::deploy()'
  on:
    all_branches: true
```

With the current default we have to restrict deployment for certain branches only in `tic.R`.

However, I find it practically easier to do it in `.travis.yml` than writing `ci()$get_branch() == "<branch>"` in `tic.R`.

I know we don't want to touch `.travis.yml` anymore when using tic but isn't this in this case complicating things? 

This leads to the general question whether conditioning (only for the deploy stage) is not easier done in `.travis.yml` than in tic.R?

Only because of the "provider, condition" interface for this stage.

I find

```yml
deploy:
  provider: script
  script: R -q -e 'tic::deploy()'
  on:
    branch: master
    condition:
      - $TRAVIS_PULL_REQUEST = false
      - $TRAVIS_EVENT_TYPE != cron
```

way simpler to understand than 

```r
if (Sys.getenv("TRAVIS_PULL_REQUEST") == "false" && 
     Sys.getenv("TRAVIS_PULL_REQUEST") != "cron" && 
       `ci()$is_branch()` == "master") {
}
```

### PR DESCRIPTION
**Pull request by pat-s**
Fixes #76 

* add `ci()$has_env()` 

* add `ci()$is_env()`

`ci_` functions are tackled in a follow-up PR.

### CODE REVIEW

**Comment by krlmlr** [comment]
Deploy preview for *wonderful-volhard-261e08* ready!

Built with commit ffe8584d07a07283e2a35381e0d8811009069d97

https://deploy-preview-124--wonderful-volhard-261e08.netlify.com

File: R/ci.R [hunk 1]
```diff
@@ -39,6 +39,15 @@ CI <- R6Class(
...
       stop("NYI")
     },
+    #'  \item{`is_env()`}{Is the given env variable set to the given value?}
+
+    is_env = function(env, value) {
+      Sys.getenv(env) == value
+    },
+    #'   \item{`has_env()`}{Does the given env variable exist?}
+    has_env = function(env) {
+      stop("NYI")
```
Comment (krlmlr) [inline R/ci.R]: Is this intended?

File: R/local.R [hunk 2]
```diff
@@ -24,6 +24,12 @@ LocalCI <- R6Class(
     get_commit = function() {
       git2r::revparse_single(revision = "HEAD")$sha
     },
+    is_env = function(env, value) {
```
Comment (krlmlr) [inline R/local.R]: Do we even need implementations in derived classes? I wonder if this should even be part of the `CI` class, maybe a simple exported function `has_env()` is enough?

File: R/ci.R (hunk 1 above)
Comment (pat-s) [reply R/ci.R]: I did not really know how to set the default correctly here as I do not really understand what you mean by "NYI" and if there is ever a situation in which this setting applies over the `ci_` subclasses? Feel free to change it :)

File: R/local.R (hunk 2 above)
Comment (pat-s) [reply R/local.R]: This is a general issue which would make all derived subclasses superfluous?

**Comment by krlmlr** [comment]
Thanks! I'll tweak.

**Pull request merged**